feat(python-lsp)!: replace the old pylsp with zuban and ruff

Cloud0310 opened the pull request:
> [!NOTE]
> See previous PR #1489 and #1172 for previous details. 
> Due to pyrefly is still not reached beta yet, we choose to wait its first beta release, as it doesn't have any comlicated refactor capibility for now.
> The PR now is converted into a draft now, the information about pyrefly could retrived [here](https://github.com/facebook/pyrefly/milestone/4) and [facebook/pyrefly#344](https://github.com/facebook/pyrefly/issues/344). And this would likely to be a **BREAKING CHANGE** once merged.

## Summary
As we are searching for a better Python LSP, and there has been updates on pyrefly as a better choice for this, which is now mature and solid, I think its time to move on for pyrefly as the default python lsp.

## Details
1. deleted: previous dependencies for pylsp, remove extra config
2. added: pyrefly (main python lsp) and ruff (format and lint) and keep the default behavior backward compatible.

Cloud0310 committed:
feat(lsp): replace the old pylsp with pyrefly and ruff

See [previous PR](https://github.com/ayamir/nvimdots/pull/1489) for
details

Cloud0310 committed:
fix: enable ruff formatter

See #1489
[comment](https://github.com/ayamir/nvimdots/pull/1489#issuecomment-3384320727)

Cloud0310 committed:
chore(linter): remove used var is_windows

Cloud0310 committed:
added(python): align with previous ruff config.

As mentioned in #1528, keep the ruff config align with previous config.

This includes:
1. set default line length to 88, as `black`'s default behavior
2. align with previous linter config

As for the pyrefly lsp config, I guess the default is good to go.

Cloud0310 committed:
chore(linter): fix line length and align with other configs

Cloud0310 committed:
Merge branch 'ayamir:main' into main

Cloud0310 committed:
Merge branch 'ayamir:main' into main

Cloud0310 commented:
Now there's another option for Python LSP, [Zuban](https://zubanls.com/) claims to support more lsp features, and look promising. We should take one that into consideration as well.

Cloud0310 committed:
Merge branch 'ayamir:main' into main

Cloud0310 committed:
Merge branch 'ayamir:main' into main

Cloud0310 committed:
Merge branch 'ayamir:main' into main

charliie-dev commented:
[Announcing Pyrefly Beta](https://youtu.be/4o0RLJJ-FAo)

charliie-dev commented:
As zuban's readme stated:
> Zuban is 20–200× faster than Mypy, while using roughly half the memory and CPU compared to Ty and Pyrefly.

Mason support: 
<img width="1069" height="812" alt="image" src="https://github.com/user-attachments/assets/563e37a4-066b-4ed2-97a7-1f208c79e02d" />

<img width="696" height="874" alt="image" src="https://github.com/user-attachments/assets/15e08d61-dbf7-443d-96b6-9077b4e3ef70" />


yeah, sure. why not lol

Cloud0310 commented:
Found this amazing test set from [python/typing](https://github.com/python/typing/tree/main/conformance), I will try to re-test, and then I guess we can choose a base on this test.

Cloud0310 commented:
> > [results.html](https://github.com/user-attachments/files/23670592/results.html) I guess as the test demonstrates, zuban is better?
> 
> can we add ty to the list?

Guess nothing is working on that one. Even basic array type infer.

Cloud0310 commented:
> I also think it would be better to choose zuban now.
> 
> P.S. There are some concerns, such as zuban being a personal project of the creator of `jedi`, so there are concerns about sustainability, and pyrefly getting more attention (from repo star). But if pyrefly become better tool, we just switch to it.

As for the personal project problem, the current solution [python-lsp](https://github.com/python-lsp/python-lsp-server?tab=readme-ov-file#installation) is based on jedi as well, so I guess the community solution is absolutely fine.

Personally, I would like the solution of using zuban for now. In the long term, if there's better solution with python lsp, I guess it wouldn't be a problem to switch to it.

P.S. I happen to know one of the contributors of zuban and pyrefly, @asukaminato0721. The situation about all mentioned lsp inplementations is as [this blog](https://asukaminato.notion.site/talked-with-pyrefly-developer-27ac81c07a9c80d9a2a0fc65eb4117e9) (Simplified Chinese) suggests.

Cloud0310 committed:
changes: switch to zuban

As [PR 1528] discussion suggests, zuban is more implemented and has less
memory footprint.

Cloud0310 commented:
So maybe we could merge this? @charliie-dev @misumisumi

charliie-dev committed:
chore(core.settings): `:sort` `lsp_deps` and `server_formatting_block_list`

Signed-off-by: Charles Chiu <mail@charliie.dev>

charliie-dev reviewed (approved):
just tested on my private python repo, seems right lol
LGTM

misumisumi reviewed (approved):
LGTM

ayamir reviewed (approved):
LGTM 💯
